curvlinops/inverse.py

```diff
@@ -327,30 +337,69 @@ def _compute_inverse_factors(
             return (aaT_eigvecs, aaT_eigvals), (ggT_eigvecs, ggT_eigvals)
         else:
             damping_aaT, damping_ggT = self._compute_damping(aaT, ggT)
-            aaT_inv = (
-                None
-                if aaT is None
-                else cholesky_inverse(
-                    cholesky(
+
+            # Compute inverse of aaT via Cholesky decomposition
+            try:
+                aaT_chol = (
+                    None
+                    if aaT is None
+                    else cholesky(
                         aaT.add(
                             eye(aaT.shape[0], dtype=aaT.dtype, device=aaT.device),
                             alpha=damping_aaT,
                         )
                     )
                 )
-            )
-            ggT_inv = (
-                None
-                if ggT is None
-                else cholesky_inverse(
-                    cholesky(
+            except Exception as exception:
+                if self._retry_double_precision and aaT.dtype != float64:
+                    warn(
+                        f"Failed to compute Cholesky decomposition in {aaT.dtype} "
+                        f"precision with exception {exception}. "
+                        "Retrying in double precision..."
+                    )
+                    # Retry in double precision
+                    aaT = aaT.to(float64)
+                    aaT_chol = cholesky(
+                        aaT.add(
+                            eye(aaT.shape[0], dtype=aaT.dtype, device=aaT.device),
+                            alpha=damping_aaT,
+                        )
+                    )
+                else:
+                    raise exception
+            aaT_inv = None if aaT_chol is None else cholesky_inverse(aaT_chol)
```

Comment:
Should convert back `aaT_inv` or `aaT_chol` to the original dtype if the re-try was triggered.